[2025-07-26T11:48:34.933+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-26T11:48:34.942+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: insert_next_row_dag.insert_row manual__2025-07-26T11:48:34.548863+00:00 [queued]>
[2025-07-26T11:48:34.946+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: insert_next_row_dag.insert_row manual__2025-07-26T11:48:34.548863+00:00 [queued]>
[2025-07-26T11:48:34.946+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-26T11:48:34.953+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): insert_row> on 2025-07-26 11:48:34.548863+00:00
[2025-07-26T11:48:34.957+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62 DeprecationWarning: This process (pid=27027) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-26T11:48:34.958+0000] {standard_task_runner.py:64} INFO - Started process 27054 to run task
[2025-07-26T11:48:34.958+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'insert_next_row_dag', 'insert_row', 'manual__2025-07-26T11:48:34.548863+00:00', '--job-id', '47', '--raw', '--subdir', 'DAGS_FOLDER/simulate_realtime_insert.py', '--cfg-path', '/tmp/tmpvuk3ltao']
[2025-07-26T11:48:34.959+0000] {standard_task_runner.py:91} INFO - Job 47: Subtask insert_row
[2025-07-26T11:48:34.966+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-07-26T11:48:34.981+0000] {task_command.py:426} INFO - Running <TaskInstance: insert_next_row_dag.insert_row manual__2025-07-26T11:48:34.548863+00:00 [running]> on host 2992d32593f0
[2025-07-26T11:48:35.018+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='insert_next_row_dag' AIRFLOW_CTX_TASK_ID='insert_row' AIRFLOW_CTX_EXECUTION_DATE='2025-07-26T11:48:34.548863+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-26T11:48:34.548863+00:00'
[2025-07-26T11:48:35.019+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-26T11:48:35.045+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/simulate_realtime_insert.py:35 SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
[2025-07-26T11:48:35.046+0000] {logging_mixin.py:188} INFO - Row 4 to insert:
[2025-07-26T11:48:35.047+0000] {logging_mixin.py:188} INFO - | user_id                      | parent_asin   |   rating | timestamp                  | main_category   | title                                                                                                                         | description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | categories                                                                                      |   price | source   |
|:-----------------------------|:--------------|---------:|:---------------------------|:----------------|:------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------|--------:|:---------|
| AEJW5NDA3GO6YBTMJBZTSYOZSPKQ | B00NQQTZCO    |        5 | 2022-06-16 14:22:32.704000 | Toys & Games    | MONOPOLY Deal Card Game, Quick-Playing Card Game for 2-5 Players, Game for Families and Kids Ages 8 and Up (Amazon Exclusive) | {"Product Description", "Get a handy way to play the classic property-trading game! Be the first collect 3 full property sets of different colors, and you'll win the Monopoly deal card game. You'll pick up cards when it's your turn and play action cards to charge players rent, steal their cards or demand money for your birthday. Build up property sets, gather piles of money and keep wheeling and dealing until you're the Monopoly deal winner! monopoly and all related characters are trademarks of Hasbro.", "From the Manufacturer", "The fast-paced, addictive card game where your luck can change in the play of a card! Collect 3 complete property sets, but beware of Debt collectors, Forced Deals and the dreaded Deal Breakers, which could change your fortunes at any time! Fun version of Monopoly for ages 8 and up!"} | {"Toys & Games", "Games & Accessories", "Card Games", "Collectible Card Games", "Decks & Sets"} |    7.69 | val      |
[2025-07-26T11:48:36.150+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-26T11:48:36.151+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "new_reviews_pkey"
DETAIL:  Key (user_id, "timestamp", price)=(AEJW5NDA3GO6YBTMJBZTSYOZSPKQ, 2022-06-16 14:22:32.704, 7.69) already exists.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/simulate_realtime_insert.py", line 43, in insert_one_row
    row_df.to_sql(
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/generic.py", line 3008, in to_sql
    return sql.to_sql(
           ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/sql.py", line 788, in to_sql
    return pandas_sql.to_sql(
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/sql.py", line 1958, in to_sql
    total_inserted = sql_engine.insert_records(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/sql.py", line 1507, in insert_records
    raise err
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/sql.py", line 1498, in insert_records
    return table.insert(chunksize=chunksize, method=method)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/sql.py", line 1059, in insert
    num_inserted = exec_insert(conn, keys, chunk_iter)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/sql.py", line 967, in _execute_insert_multi
    result = conn.execute(stmt)
             ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1385, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1953, in _execute_context
    self._handle_dbapi_exception(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 2134, in _handle_dbapi_exception
    util.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1910, in _execute_context
    self.dialect.do_execute(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "new_reviews_pkey"
DETAIL:  Key (user_id, "timestamp", price)=(AEJW5NDA3GO6YBTMJBZTSYOZSPKQ, 2022-06-16 14:22:32.704, 7.69) already exists.

[SQL: INSERT INTO public.new_reviews (user_id, parent_asin, rating, timestamp, main_category, title, description, categories, price, source) VALUES (%(user_id_m0)s, %(parent_asin_m0)s, %(rating_m0)s, %(timestamp_m0)s, %(main_category_m0)s, %(title_m0)s, %(description_m0)s, %(categories_m0)s, %(price_m0)s, %(source_m0)s)]
[parameters: {'user_id_m0': 'AEJW5NDA3GO6YBTMJBZTSYOZSPKQ', 'parent_asin_m0': 'B00NQQTZCO', 'rating_m0': 5.0, 'timestamp_m0': datetime.datetime(2022, 6, 16, 14, 22, 32, 704000), 'main_category_m0': 'Toys & Games', 'title_m0': 'MONOPOLY Deal Card Game, Quick-Playing Card Game for 2-5 Players, Game for Families and Kids Ages 8 and Up (Amazon Exclusive)', 'description_m0': '{"Product Description", "Get a handy way to play the classic property-trading game! Be the first collect 3 full property sets of different colors, an ... (527 characters truncated) ... bt collectors, Forced Deals and the dreaded Deal Breakers, which could change your fortunes at any time! Fun version of Monopoly for ages 8 and up!"}', 'categories_m0': '{"Toys & Games", "Games & Accessories", "Card Games", "Collectible Card Games", "Decks & Sets"}', 'price_m0': '7.69', 'source_m0': 'val'}]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2025-07-26T11:48:36.175+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=insert_next_row_dag, task_id=insert_row, run_id=manual__2025-07-26T11:48:34.548863+00:00, execution_date=20250726T114834, start_date=20250726T114834, end_date=20250726T114836
[2025-07-26T11:48:36.183+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 47 for task insert_row ((psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "new_reviews_pkey"
DETAIL:  Key (user_id, "timestamp", price)=(AEJW5NDA3GO6YBTMJBZTSYOZSPKQ, 2022-06-16 14:22:32.704, 7.69) already exists.

[SQL: INSERT INTO public.new_reviews (user_id, parent_asin, rating, timestamp, main_category, title, description, categories, price, source) VALUES (%(user_id_m0)s, %(parent_asin_m0)s, %(rating_m0)s, %(timestamp_m0)s, %(main_category_m0)s, %(title_m0)s, %(description_m0)s, %(categories_m0)s, %(price_m0)s, %(source_m0)s)]
[parameters: {'user_id_m0': 'AEJW5NDA3GO6YBTMJBZTSYOZSPKQ', 'parent_asin_m0': 'B00NQQTZCO', 'rating_m0': 5.0, 'timestamp_m0': datetime.datetime(2022, 6, 16, 14, 22, 32, 704000), 'main_category_m0': 'Toys & Games', 'title_m0': 'MONOPOLY Deal Card Game, Quick-Playing Card Game for 2-5 Players, Game for Families and Kids Ages 8 and Up (Amazon Exclusive)', 'description_m0': '{"Product Description", "Get a handy way to play the classic property-trading game! Be the first collect 3 full property sets of different colors, an ... (527 characters truncated) ... bt collectors, Forced Deals and the dreaded Deal Breakers, which could change your fortunes at any time! Fun version of Monopoly for ages 8 and up!"}', 'categories_m0': '{"Toys & Games", "Games & Accessories", "Card Games", "Collectible Card Games", "Decks & Sets"}', 'price_m0': '7.69', 'source_m0': 'val'}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 27054)
[2025-07-26T11:48:36.213+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-07-26T11:48:36.233+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-26T11:48:36.234+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
